src/components/EmptyState/EmptyState.jsx

```diff
@@ -0,0 +1,130 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import Button from '../Button';
+import { Link } from '../Link';
+import { settings } from '../../constants/Settings';
+import {
+  EmptystateErrorIcon as ErrorImage,
+  Emptystate404Icon as Error404Image,
+  EmptystateDefaultIcon as EmptyImage,
+  EmptystateSuccessIcon as SuccessImage,
+  EmptystateNoresultsIcon as NoResultImage,
+  EmptystateNotauthorizedIcon as NotAuthImage,
+} from '../../icons/components';
+
+const { iotPrefix } = settings;
+
+const icons = {
+  error: ErrorImage,
+  error404: Error404Image,
+  empty: EmptyImage,
+  'not-authorized': NotAuthImage,
+  'no-result': NoResultImage,
+  success: SuccessImage,
+};
+
+const props = {
+  /** Title of empty state */
+  title: PropTypes.string.isRequired,
+  /** Description of empty state */
+  body: PropTypes.string.isRequired,
+  /** Optional image of state */
+  icon: PropTypes.oneOfType([
+    PropTypes.func,
+    PropTypes.oneOf([
+      'error',
+      'error404',
+      'empty',
+      'not-authorized',
+      'no-result',
+      'success',
+      '',
+    ]),
+  ]),
+  /** Optional action for container */
+  action: PropTypes.shape({
+    label: PropTypes.string.isRequired,
+    onClick: PropTypes.func.isRequired,
+  }),
+  /** Optional secondary action for container */
+  secondaryAction: PropTypes.shape({
+    label: PropTypes.string.isRequired,
+    onClick: PropTypes.func.isRequired,
+  }),
+  /** Specify an optional className to be applied to the container */
+  className: PropTypes.string,
+  /** Specify a testid for testing this component */
+  testID: PropTypes.string,
+};
+
+const defaultProps = {
+  action: null,
+  secondaryAction: null,
+  icon: '',
+  className: '',
+  testID: 'EmptyState',
+};
+
+/**
+ * Component to set empty states
+ * For reference, visit https://pages.github.ibm.com/ai-applications/design/components/empty-states/usage/
+ */
+const EmptyState = ({
+  title,
+  icon,
+  body,
+  action,
+  secondaryAction,
+  className,
+  testID,
+}) => (
+  <div
+    className={`${iotPrefix}--empty-state ${className}`}
+    data-testid={testID}>
+    <div className={`${iotPrefix}--empty-state--content`}>
+      {icon &&
+        React.createElement(typeof icon === 'string' ? icons[icon] : icon, {
+          className: `${iotPrefix}--empty-state--icon`,
+          alt: '',
+          'data-testid': `${testID}-icon`,
+        })}
+      <h3
+        className={`${iotPrefix}--empty-state--title`}
+        data-testid={`${testID}-title`}>
+        {title}
+      </h3>
+      <p
+        className={`${iotPrefix}--empty-state--text`}
+        data-testid={`${testID}-body`}>
+        {body}
+      </p>
+      {action && (
+        <div
+          className={`${iotPrefix}--empty-state--action`}
+          data-testid={`${testID}-action`}>
+          <Button onClick={action.onClick && action.onClick}>
```

Comment:
Why `action.onClick && action.onClick`? 
(We know there is an action and if there is an action then the `onClick` prop is required)